Task1/todo.js

```diff
@@ -0,0 +1,134 @@
+// This is the array that will hold the todo list items
+let todoItems = [];
+// This function will create a new todo object with text input, 
+//check to see which task is completed and id for remove
+
+// Render the todo items
+function renderTodo(todo)
+{
+    const list=document.querySelector("#todoList");
+    //select item which todo-id matches
+    const item=document.querySelector(`[todo-id='${todo.id}']`);
+    //delete the item
+    if(todo.deleted)
+    {
+        item.remove();
+        return
+    }
+
+    const node=document.createElement("li");
+    //create attribute as it uniquely define the key
+    node.setAttribute('todo-id',todo.id);
+    //creating delelte btn
+    const deleteButton =document.createElement("button");
+    deleteButton.classList.add('js-delete-todo');
+    deleteButton.innerHTML="Delete" 
+    //creating span 
+    node.innerHTML = `
+    <span>${todo.text}</span>
+    `;
+    //appending node and btn
+    node.appendChild(deleteButton)
+    list.appendChild(node)
+}
+
+//function to add item in an array
+function addTodo(text)
+{
+    const todo={
+        text,
+        id:Date.now(),
+    };
+
+    todoItems.push(todo);
+    //changed this 
+    todoItems.forEach((todo)=>{
+        renderTodo(todo);
+    });
+    // renderTodo(todo);
+}
+
+
+
+//function to delete each item
+function deleteTodo(idToBeRemoved)
+{
+
+    const index=todoItems.findIndex(({id})=>id===Number(idToBeRemoved));
+    //create an object 
+    const todo={
+        deleted:true,
+        ...todoItems[index]
+    };
+    //create an array after removing the items 
+    todoItems=todoItems.filter(({id})=>id!==Number(idToBeRemoved));
+    renderTodo(todo);
+}
+
+
+
+//Select the form element
+const form = document.querySelector("#todoForm");
+//Add event listener
+const add = document.querySelector("#button");
+add.addEventListener('click',event=>{
+    //prevent page from refresh on clicking
+    event.preventDefault();
+    //select the input
+    const input=document.querySelector("#todoInput");
+    //get value of input and remove whitespaces we can use createNodetext also
+    const text=input.value.trim();
+    //updated from text!=='' to text 
+    if(text)
+    {
+        addTodo(text);
+        //reset value after pushing to ''
+        input.value='';
+        //give focus to element
+        input.focus();
+    }
+});
+
+
+
+//selecting the delete button
+const list=document.querySelector('#todoList');
+list.addEventListener('click',event=>{
+    if(event.target.classList.contains('js-delete-todo'))
+    {
+        const itemKey=event.target.parentElement;
+        deleteTodo(itemKey.getAttribute('todo-id'));
+        //getattribute gives string in retrun
+    }
+});
+
+
+//function to delelte all items
+const deleteall=document.querySelector('.delete-all');
```

Comment:
1. Variables naming should always be camelCase: `const deleteAllTodos`
2. Another way to check for array length should be greater than 0 can be: `while(!!todoItems.length) { }`

@Poorvi-3pillar